Elaborate on support for hash helpers in Integration Tests [ci skip] (#50289)
rails/rails: guides/source/testing.md

### The Three Hashes of the Apocalypse

**After** a request has been made and processed, you will have 3 Hash objects ready for use:

* `cookies` - Any cookies that are set
* `flash` - Any objects living in the flash
* `session` - Any object living in session variables

```ruby
test "POST /sessions signs in a User" do
  alice = users(:alice)

  post sign_in_path, params: { email: alice.email, password: alice.password }

  assert_equal alice.id, session[:user_id]
  assert_equal "Successfully signed in.", flash[:notice]
end
```

> As is the case with normal Hash objects, you can access the values by referencing the keys by String. You can also reference them by Symbol name. For example:

Reviewer: I'm not sure the current style for constants, I feel like we normally don't capitalize Hash/String/Symbol/etc unless it is a Rails constant "ActiveRecord::Base", or "an Active Record object". 🤔

```ruby
flash["gordon"]               # or flash[:gordon]
session["shmession"]          # or session[:shmession]
cookies["are_good_for_u"]     # or cookies[:are_good_for_u]
```

> NOTE: Writing to these values from within the test case isn't supported. Reading from these values prior to the initial request also isn't supported.

Reviewer: ```suggestion
NOTE: These values are read-only, and can only be read after the initial request is made.
```

> NOTE: Writing to these values from within the test case isn't supported. Reading from these values isn't prior to the initial request also isn't supported.

Reviewer 2: Did you mean:

```suggestion
NOTE: Writing to these values from within the test case isn't supported. Reading from these values prior to the initial request also isn't supported.
```

> If your test requires values to be set, make a request that will set that value. Requests made throughout a test case (outside of an [`open_session`](https://api.rubyonrails.org/classes/ActionDispatch/Integration/Runner.html#method-i-open_session) block) will share a single Rack session.

Reviewer: ```suggestion
If your test requires writing to either of these values, make a request that will set the desired value in one of `cookies`, `flash`, or `session`. Requests made during a test case (outside of an [`open_session`](https://api.rubyonrails.org/classes/ActionDispatch/Integration/Runner.html#method-i-open_session) block) will share a single Rack session.
```

> If you the test requires values to be set, make a request that will set that value. Requests made throughout a test case (outside of an [`open_session`](https://api.rubyonrails.org/classes/ActionDispatch/Integration/Runner.html#method-i-open_session) block) will share a single Rack session.

Reviewer 2: ```suggestion
If your test requires values to be set, make a request that will set that value. Requests made throughout a test case (outside of an [`open_session`](https://api.rubyonrails.org/classes/ActionDispatch/Integration/Runner.html#method-i-open_session) block) will share a single Rack session.
```

> For example, if a test requires a `session[:user_id]` value set during sign in, and successfully signing in writes to that value, make a request to sign in prior to the action under test:

Reviewer: ```suggestion
For example, if a test assumes a successful sign in writes to `session[:user_id]`, make a request to sign in prior to the action under test:
```
